src/utils/Math.ts

```diff
@@ -1,3 +1,7 @@
-export function clamp(input: number, lower: number, upper: number): number {
+export const clamp = (input: number, lower: number, upper: number): number => {
   return Math.min(Math.max(input, lower), upper);
-}
+};
+
+export const randomNumberBetween = (min = 0, max = 1): number => {
```

Comment:
This function is only called when getting the sorting position between two items, and it seems odd to be calling a random number generator for that. I feel like using an average would give better performance.

Reply:
Maybe if I go with the solution of storing integers as we discussed on the backend part I wont need this. I am thinking of going back to integers and update all the later items in the list.